File: lez09_dadi/public_html/Giocatore.js
export default class Giocatore{
       
       constructor(soldi,nome){ 
           if(soldi < 0){
               throw "soldi negativi non validi";
           }
           this._soldi = soldi;
           this.nome = nome;
       } 
       
       incrementa(){
           this.soldi++;
       }
       
       decrementa(){
           this.soldi--;
       }
       
       get soldi(){
           return this._soldi;
       }
       
       set soldi(valore){
           if(valore < 0){
               throw "soldi negativi non validi";
           }
           this._soldi= valore;
       }
}

File: lez09/src/dadi/Giocatore.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package dadi;

public class Giocatore {

    private int soldi;
    private String nome;

    public Giocatore(int valore, String nome) {
        if (valore <= 0) {
            throw new IllegalArgumentException("I soldi non possono essere negativi");
        }
        soldi = valore;
        this.nome = nome;
    }

    public void incrementa() {
        soldi++;
    }

    public void decrementa() {
        soldi--;
    }

    public int getSoldi() {
        return soldi;
    }

    public String getNome() {
        return nome;
    }

}